master/internal/db/postgres_rp_workspace_bindings.go

```diff
@@ -77,7 +77,7 @@ func ReadWorkspacesBoundToRP(
 ) ([]*RPWorkspaceBinding, *apiv1.Pagination, error) {
 	var rpWorkspaceBindings []*RPWorkspaceBinding
 	query := Bun().NewSelect().Model(&rpWorkspaceBindings).Where("pool_name = ?",
-		poolName)
+		poolName).Where("valid = ?", true)
```

Comment:
I thought about this again after the discussion, so I can understand it better. 

If a RP is invalid, the API returns an error.

In the case that RP A has bindings with WS1, WS2, and WS3. Is there a case we plan on marking some of these bindings are invalid? If so, then I think this WHERE clause is worthwhile, so the user only get valid bindings. Ideally, it will be more helpful if the user can see which ones are valid and which ones are invalid, but our design needs to be changed. The current design return a list of workspaces.

You may already explained this, but I didn't fully understand during the discussion. Sorry for the repeated question.

Reply:
yeah case 2 shouldn't ever happen because we will invalidate on a per resource pool basis, so I think it's fine to omit the WHERE clause.